File: 3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        res = []
        for num in nums:
            for a in range(num):
                if a | (a + 1) == num:
                    res.append(a)
                    break
            
            else:
                res.append(-1)
        
        return res
            
            

File: 3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.go
func minBitwiseArray(nums []int) []int {
	result := make([]int, len(nums))

	for i, original := range nums {
		candidate := -1

		for j := 1; j < original; j++ {
			if (j | (j + 1)) == original {
				candidate = j
				break
			}
		}

		result[i] = candidate
	}

	return result
}